test/e2e/operator.go

```diff
@@ -419,3 +420,131 @@ var _ = Describe("ARO Operator - MHC Deployment", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Image Reconciler", func() {
+	imageconfigFlag := "aro.imageconfig.enabled"
+	optionalRegistry := "quay.io"
+	ctx := context.Background()
+	timeout := 5 * time.Minute
+	var expectedAllowlist, expectedBlocklist []string
+
+	getLocationRegistries := func(acrDomain, acrDnsSuffix, location string) []string {
+		regionalAcr := fmt.Sprintf("arosvc.%s.data.%s", location, acrDnsSuffix)
+		requiredRegistries := []string{acrDomain, regionalAcr}
+		return requiredRegistries
+	}
+
+	sliceEqual := func(a, b []string) bool {
+		if len(a) != len(b) {
+			return false
+		}
+		sort.Strings(a)
+		sort.Strings(b)
+
+		for idx, entry := range b {
+			if a[idx] != entry {
+				return false
+			}
+		}
+		return true
+	}
+
+	verifyLists := func() (bool, error) {
+		imageconfig, err := clients.ConfigClient.ConfigV1().Images().Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		allowList := imageconfig.Spec.RegistrySources.AllowedRegistries
+		blockList := imageconfig.Spec.RegistrySources.BlockedRegistries
+
+		return sliceEqual(allowList, expectedAllowlist) && sliceEqual(blockList, expectedBlocklist), nil
+	}
+
+	AfterEach(func() {
+		// Clear allow/block lists
+		imageconfig, err := clients.ConfigClient.ConfigV1().Images().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		imageconfig.Spec.RegistrySources.AllowedRegistries = nil
+		imageconfig.Spec.RegistrySources.BlockedRegistries = nil
+		expectedAllowlist = nil
+		expectedBlocklist = nil
+
+		_, err = clients.ConfigClient.ConfigV1().Images().Update(ctx, imageconfig, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(verifyLists).WithTimeout(timeout).Should(BeTrue())
```

Comment:
The testing framework would let you do something like `Eventually(Expect(actualImageConfig).To(BeEquivalentTo(expectedImageConfig))).WithTimeout(timeout)`. If it works, I think it might make more sense than the functions to assert that they match.